build-system/tasks/extension-generator/bento/amp-__component_name_hyphenated__/__component_version__/amp-__component_name_hyphenated__.js

```diff
@@ -14,18 +14,16 @@
  * limitations under the License.
  */
 
-import {CSS as COMPONENT_CSS} from './__component_name_hyphenated__.jss';
+import {BaseElement} from './base-element';
 import {CSS} from '../../../build/amp-__component_name_hyphenated__-__component_version__.css';
-import {__component_name_pascalcase__} from './__component_name_hyphenated__';
-import {PreactBaseElement} from '../../../src/preact/base-element';
 import {dict} from '../../../src/utils/object';
 import {isExperimentOn} from '../../../src/experiments';
 import {userAssert} from '../../../src/log';
 
 /** @const {string} */
 const TAG = 'amp-__component_name_hyphenated__';
 
-class Amp__component_name_pascalcase__ extends PreactBaseElement {
+class Amp__component_name_pascalcase__ extends BaseElement {
   /** @override */
   init() {
```

Comment:
Could the `return` part of `init` be moved to the `BaseElement` definition? And the `registerApiAction` would stay here in the AMP layer?

Reply:
I do not know the answer to this. I think it can as long as it doesn't depend on any amp-only properties or methods, but I'm not certain.